File: XamarinStore.Droid/Fragments/BragFragment.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Util;
using Android.Views;
using Android.Widget;

namespace XamarinStore
{
	public class BragFragment : Fragment
	{
		public override void OnCreate (Bundle savedInstanceState)
		{
			base.OnCreate (savedInstanceState);
			RetainInstance = true;
			// Create your fragment here
		}

		public override View OnCreateView (LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState)
		{
			var view = inflater.Inflate (Resource.Layout.BragScreen,null);
			var btn = view.FindViewById<Button> (Resource.Id.bragButton);
			btn.Click += (object sender, EventArgs e) => {
				BragOnTwitter();
			};
			return view;
		}

		void BragOnTwitter()
		{
			var message = "";
			try{
				var intent = new Intent(Intent.ActionSend);
				intent.PutExtra(Intent.ExtraText,message);
				intent.SetType("text/plain");
				StartActivity(Intent.CreateChooser(intent, Resources.GetString(Resource.String.brag_on)));
			} catch(Exception e) {
				Console.WriteLine (e);
			}
		}
	}
}

File: F#/XamarinStore.Droid/Fragments/BragFragment.fs
namespace XamarinStore

open System
open Android.App
open Android.Content
open Android.OS
open Android.Views
open Android.Widget

type BragFragment() as this =
    inherit Fragment()

    let BragOnTwitter() =
        let message = ""
        try
            let intent = new Intent(Intent.ActionSend)
            intent.PutExtra(Intent.ExtraText,message) |> ignore
            intent.SetType("text/plain") |> ignore
            this.StartActivity(Intent.CreateChooser(intent, this.Resources.GetString(Resource_String.brag_on)))
        with e-> Console.WriteLine e

    override this.OnCreate savedInstanceState =
        base.OnCreate savedInstanceState
        this.RetainInstance <- true

    override this.OnCreateView (inflater, container, savedInstanceState) =
        let view = inflater.Inflate (Resource_Layout.BragScreen, null)
        let btn = view.FindViewById<Button> Resource_Id.bragButton
        btn.Click.Add(fun _ -> BragOnTwitter() )
        view
